RELEASE_NOTES.rst

```diff
@@ -21,6 +21,279 @@
 
 .. towncrier release notes start
 
+Airflow 2.5.0 (2022-11-28)
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+- ``airflow dags test`` no longer performs a backfill job.
+
+  In order to make ``airflow dags test`` more useful as a testing and debugging tool, we no
+  longer run a backfill job and instead run a "local task runner". Users can still backfill
+  their DAGs using the ``airflow dags backfill`` command. (#26400)
+- Airflow config section ``kubernetes`` renamed to ``kubernetes_executor``
+
+  KubernetesPodOperator no longer considers any core kubernetes config params, so this section now only applies to kubernetes executor. Renaming it reduces potential for confusion. (#26873)
+- ``ExternalTaskSensor`` no longer hangs indefinitely when ``failed_states`` is set, an ``execute_date_fn`` is used, and some but not all of the dependent tasks fail. Instead, an ``AirflowException`` is thrown as soon as any of the dependent tasks fail.
+
+  Any code handling this failure in addition to timeouts should move to cathing the ``AirflowException`` baseclass and not only the ``AirflowSensorTimeout`` subclass. (#27190)
```

Comment:
This will need a similar change as above, but we probably need a shorter title for this?